backend/routes/paypal.js

```diff
@@ -0,0 +1,233 @@
+        })
+      }
+
+      const client = getClient(paypalClientId, paypalClientSecret)
+
+      const request = new PayPal.orders.OrdersCreateRequest()
+      request.requestBody({
+        intent: 'CAPTURE',
+        purchase_units: [
+          {
+            amount: {
+              currency_code: (currency || 'USD'),
+              value: req.body.amount
+            },
+            // PayPal doesn't support metadata,
+            // so using invoice_id and custom_id instead
+            invoice_id: randomstring.generate(), // paymentCode
+            custom_id: req.body.data // encryptedData IPFS hash
+          }
+        ],
+        application_context: {
+          user_action: 'PAY_NOW',
+          return_url: req.body.returnUrl,
+          cancel_url: req.body.cancelUrl
+        }
+      })
+
+      const response = await client.execute(request)
+
+      log.debug(response.result)
+
+      const authorizeUrl = response.result.links.find(
+        (l) => l.rel === 'approve'
+      ).href
+
+      res.send({
+        success: true,
+        authorizeUrl,
+        orderId: response.result.id
+      })
+    } catch (err) {
+      log.error(`[Shop ${req.shop.id}] Failed to create order on PayPal`, err)
+      res.send({
+        success: false
+      })
+    }
+  })
+
+  router.post('/paypal/capture', authShop, async (req, res) => {
+    try {
+      const network = await Network.findOne({
+        where: { networkId: req.shop.networkId }
+      })
+
+      const networkConfig = getConfig(network.config)
+      const shopConfig = getConfig(req.shop.config)
+      const web3Pk = shopConfig.web3Pk || networkConfig.web3Pk
+      
+      const { clientId } = req.body
+      const { paypalClientSecret, paypalClientId } = shopConfig
+
+      if (!web3Pk || !paypalClientSecret || !paypalClientId || clientId !== paypalClientId) {
+        return res.status(400).send({
+          success: false,
+          reason: 'PayPal payment is disabled'
+        })
+      }
+
+
+      log.info(
+        `[Shop ${req.shop.id}] Çapturing payment on PayPal`,
+        req.body.orderId
+      )
+
+      const client = getClient(paypalClientId, paypalClientSecret)
+
+      const request = new PayPal.orders.OrdersCaptureRequest(req.body.orderId)
+
+      const response = await client.execute(request)
+
+      log.debug(response.result)
+
+      res.send({
+        success: true
+      })
+    } catch (err) {
+      log.error(`[Shop ${req.shop.id}] Failed to create order on PayPal`, err)
+      res.send({
+        success: false
+      })
+    }
+  })
+
+  const findShop = async (req, res, next) => {
+    const { shopId } = req.params
+    const shop = await Shop.findOne({ where: { id: shopId } })
+    req.shop = shop
+    next()
+  }
+
+  const webhookHandler = async (req, res, next) => {
+    const { shopId } = req.params
+
+    try {
+      const event = req.body
+
+      const externalPayment = await ExternalPayment.create({
```

Comment:
Couple comments:
 - Why calling ExternalPayment.create then afterwards calling save? This results in 2 writes to the database. A single call to create() with all the data should suffice?
 - It would likely be helpful to be able to differentiate between Stripe and PayPal rows when querying the data. So might be good to add a new column "processor" which would be an ENUM with currently 2 possible values: Stripe or PayPal.